esp-hal/src/sha.rs

```diff
@@ -126,12 +279,31 @@ impl crate::interrupt::InterruptConfigurable for Sha<'_> {
 /// see ::finish() length/self.cursor usage
 pub struct ShaDigest<'d, A, S: Borrow<Sha<'d>>> {
     sha: S,
+    state: DigestState,
+    phantom: PhantomData<(&'d (), A)>,
+}
+
+#[derive(Clone, Debug)]
+struct DigestState {
+    algorithm: ShaAlgorithmKind,
     alignment_helper: AlignmentHelper<SocDependentEndianess>,
     cursor: usize,
     first_run: bool,
     finished: bool,
     message_buffer_is_full: bool,
```

Comment:
I wish Rust had a built in way to tightly pack these fields into a single `u16`.